src/main/java/com/esri/samples/na/routing_around_barriers/RoutingAroundBarriersController.java

```diff
@@ -0,0 +1,331 @@
+            Route firstRoute = routeResult.getRoutes().get(0);
+
+            // create a geometry for this route
+            Geometry routeGeometry = firstRoute.getRouteGeometry();
+
+            // create a graphic for the route and add it to the graphics overlay
+            Graphic routeGraphic = new Graphic(routeGeometry, routeLineSymbol);
+            routeGraphicsOverlay.getGraphics().add(routeGraphic);
+
+            // set the title of the TitledPane to display the information
+            String output = String.format("Route directions: %d min (%.2f km)", Math.round(firstRoute.getTravelTime()), firstRoute.getTotalLength() / 1000);
+            routeInformationTitledPane.setText(output);
+
+            // get the direction text for each maneuver and add them to the list to display
+            for (DirectionManeuver maneuver : firstRoute.getDirectionManeuvers()) {
+              directionsList.getItems().add(maneuver.getDirectionText());
+            }
+
+          } else {
+            new Alert(Alert.AlertType.ERROR, "No possible routes found").show();
+          }
+
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Solve RouteTask failed " + e.getMessage() + e.getMessage()).show();
+        }
+
+        // enable the reset button
+        btnReset.setDisable(false);
+
+      });
+    } else {
+      new Alert(Alert.AlertType.ERROR, "Cannot run the routing task, a minimum of two stops is required").show();
+    }
+  }
+
+  /**
+   * Clears stops and barriers from the route parameters, clears direction list and graphics overlays.
+   */
+  @FXML
+  private void clearRouteAndGraphics() {
+    // clear stops from route parameters and stops list
+    routeParameters.clearStops();
+    stopsList.clear();
+
+    // clear barriers from route parameters and barriers list
+    routeParameters.clearPointBarriers();
+    barriersList.clear();
+
+    // reset the route information title pane
+    routeInformationTitledPane.setText("No route to display");
+
+    // clear the directions list
+    directionsList.getItems().clear();
+
+    // clear all graphics overlays
+    mapView.getGraphicsOverlays().forEach(graphicsOverlay -> graphicsOverlay.getGraphics().clear());
+
+    // enable/disable buttons
+    btnDetermineRoute.setDisable(false);
+    btnReset.setDisable(true);
+  }
+
+  /**
+   * Builds a composite symbol out of a pin symbol and a text symbol marking the stop number.
+   *
+   * @param stopNumber The number of the current stop being added
+   * @return A composite symbol to mark the current stop
+   */
+  private CompositeSymbol createCompositeStopSymbol(Integer stopNumber) {
+    // determine the stop number and create a new label
+    TextSymbol stopTextSymbol = new TextSymbol(16, (stopNumber).toString(), 0xFFFFFFFF, TextSymbol.HorizontalAlignment.CENTER, TextSymbol.VerticalAlignment.BOTTOM);
+    stopTextSymbol.setOffsetY((float) pinImage.getHeight() / 2);
```

Comment:
The pin symbol may need an offset too because it is not placed where you click.

Reply:
Can you share a screenshot/gif perhaps? On my machine it seems to be directly where clicked

![98725532-C20D-4510-8EF6-B3C93EB45E48](https://user-images.githubusercontent.com/44193148/61202738-427ec480-a6e0-11e9-8a05-2b45da4c4bfc.GIF)